pandas/core/algorithms.py

```diff
@@ -343,6 +344,8 @@ def factorize(values, sort=False, order=None, na_sentinel=-1, size_hint=None):
             vals = values.view(np.int64)
     else:
         vals = np.asarray(values)
+        if is_bool_dtype(vals):
+            vals = vals.astype(int)
```

Comment:
I think you will have to add there `dtype = bool`? (or `dtype = values.dtype)` 
Otherwise the uniques will also be 0/1